### DISCUSSION TITLE
Why didn't you use rust?

### DISCUSSION THREAD

**Discussion by ghost** [nats-io/nats-server#4908] [Q&A]
why ???????????

**Comment by derekcollison** [comment]
Rust as a language did not exist. And even if it did, I would not have chosen it to develop the 2nd generation NATS server.

**Reply by ghost** [reply]
why you don't want to choose rust for 2nd generation NATS server .. can you explain it ??

**Reply by derekcollison** [reply]
The current server is the 2nd generation server, it was started in 2012. Rust was not a well known language at that time and was still very young.

**Reply by ghost** [reply]
we can rewrite it in rust ... ??? I think NATS is a nice messaging system..  I really  like it

**Reply by derekcollison** [reply]
If we decide to do a 3rd generation server, which would be a massive undertaking, it would not be in Rust. Current best candidate would be Zig.

**Reply by Jarema** [reply]
Aside from discussing future of nats-server, Rust users are first class citizens in NATS ecosystem (and many other languages are too) with officially supported Rust Client https://github.com/nats-io/nats.rs . So the fact that the server is written in different language should not make it any harder to adopt if you like it!

**Reply by ripienaar** [reply]
Can you list a set of benefits that would make the multi years and realistically millions of dollars  in cost worth while? Consider also lost opportunity cost while other new features could be added in the same time instead.

**Reply by ghost** [reply]
why Zig ??/ please explain it?? @derekcollison

**Reply by derekcollison** [reply]
We have made no definitive plans on a 3rd generation server, so this would be premature. For the foreseeable future the server will remain in Go.

**Reply by ghost** [reply]
I know but why you would write in Zig over Rust
